packages/core/core/src/Environment.js

```diff
@@ -97,3 +98,14 @@ export function mergeEnvironments(
     ...b
   });
 }
+
+export function getEnvironmentHash(env: Environment) {
+  // context is excluded from hash so that assets can be shared between e.g. workers and browser.
+  // Different engines should be sufficient to distinguish multi-target builds.
```

Comment:
I'm not 100% sure about doing this. It's theoretically possible for a transformer to return different content if `env.context === 'web-worker'` for example, and this effectively prevents that. The benefits of sharing assets between web workers and main thread are pretty compelling though. I guess the only other option is to remove workers as separate contexts from browser...